tests/functional/test_credentials.py

```diff
@@ -789,7 +790,8 @@ def assert_session_credentials(self, expected_params, **kwargs):
         expected_creds = self.create_random_credentials()
         response = self.create_assume_role_response(expected_creds)
         session = StubbedSession(**kwargs)
-        stubber = session.stub('sts')
+        config = Config(signature_version=UNSIGNED)
+        stubber = session.stub('sts', config=config)
```

Comment:
Without this change `test_assume_role` would always get stuck. I believe it's because the stubber doesn't take full effect until the `before-call` event is emitted, which isn't until after endpoint resolution. So the `assume_role_with_web_identity` call would now only partially be mocked.

Adding this config setting resolved the issue and I believe is an acceptable change because the provider uses `UNSIGNED` in actuality https://github.com/boto/botocore/blob/develop/botocore/credentials.py#L904

Reply:
After coming back to this, I'm not seeing the test hanging anymore. I think it must've been something wrong with my local environment. I'll revert the change and keep an eye on it in CI